platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -400,10 +400,17 @@ export default class HangingProtocolService extends PubSubService {
     this.displaySets = displaySets;
     this.setActiveStudyUID((activeStudy || studies[0])?.StudyInstanceUID);
 
-    this.protocolEngine = new ProtocolEngine(
-      this.getProtocols(),
-      this.customAttributeRetrievalCallbacks
-    );
+    const { customizationService } = this._servicesManager.services;
+    const customAttributeRetrievalCallbacks = customizationService.get(
+      'HPCustomAttributeRetrievalCallbacks'
```

Comment:
what is HPCustomAttributeRetrievalCallbacks?
Please add an example to the documentation of hanging protocol showing it

Reply:
It's by using the addCustomAttribute in the preRegistration hook of the extension, in case you're curious.